Task 3/system/Coding/src/app/StateHandler.c

```diff
@@ -24,13 +24,169 @@
 /**
  * @brief store the current state ID
 */
-static States gCurrentState;
+static Bool gProcessedEntryFunction = FALSE;
+
+/**
+ * @brief store the current state ID
+*/
+static States gCurrentState = INITIALIZATION_STATE;
+
+static ErrorHandlerErrorCode gErrorCode = ERRORHANDLER_MAIN_SCHEDULER_EXIT;
 
 /* EXTERNAL FUNCTIONS *****************************************************************************/
 
 void StateHandler_stateHandler(void)
 {
-   
+    switch (gCurrentState)
+    {
+        case DRIVE_LAP_STATE:
+            DriveLapState_processDriveOnTrackLine();
+
+            if (DriveLapState_checkTranstionTriggerStartlineFound())
+            {
+                gCurrentState = LAP_FINISHED_STATE;
+            }
+            else if (DriveLapState_checkTranstionTriggerTrackNotFound())
+            {
+                gCurrentState = ERROR_STATE;
+            }
+            else if (DriveLapState_checkTranstionTriggerTimer2Exceeds20s())
+            {
+                gCurrentState = ERROR_STATE;
+                gErrorCode = ERRORHANDLER_DRIVING_TIMEOUT;
+            }
```

Comment:
Missing else handling. 

This is required to ensure that every case is explicitly handled. This is also required even though in case of else nothing is done.

```suggestion
            }
            else
            {
                /* Do nothing. */
            }
```